lib/postMessage.js

```diff
@@ -1,30 +1,24 @@
 const fetch = require('node-fetch')
 const SemanticReleaseError = require('@semantic-release/error')
 
-async function postMessage(message, logger, slackWebhook) {
-	await fetch(slackWebhook, {
-		method: 'post',
-		headers: {
-			'Content-Type': 'application/json'
-		},
-		body: JSON.stringify(message)
-	})
-		.then(res => res.text())
-		.then(text => {
-			if (text !== 'ok') {
-				logger.log('JSON message format invalid: ' + text)
-				throw new SemanticReleaseError(
-					new Error().stdout,
-					'INVALID SLACK COMMAND: ' + text
-				)
-			}
+module.exports = async (message, logger, slackWebhook) => {
+	let response
+	let bodyText
+	try {
+		response = await fetch(slackWebhook, {
+			method: 'post',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify(message)
 		})
-		.catch(e => {
-			throw new SemanticReleaseError(
-				e.stdout,
-				'SLACK CONNECTION FAILED: '
-			)
-		})
-}
+		bodyText = await response.text()
+	} catch (e) {
+		throw new SemanticReleaseError(e.message, 'SLACK CONNECTION FAILED')
+	}
 
-module.exports = postMessage
+	if (!response.ok || bodyText !== 'ok') {
+		logger.log('JSON message format invalid: ' + bodyText)
+		throw new SemanticReleaseError(bodyText, 'INVALID SLACK COMMAND')
+	}
```

Comment:
Could this be moved into the try-block? If so `response` and `bodyText` could be `const`

Reply:
No. Because then we would end up in the same "problem" as before. Which was that the catch clause will, well, catch the exception being thrown and obfuscate it.

Reply:
Ah! That makes sense